syntax = "proto3";

package user;

//username is unique here
service userService{
    rpc createAccount (aUser) returns (isSuccess);
    rpc login(aUser) returns (session);
    rpc auth(header) returns (isSuccess);
}

message header{
    string token=1;
}

message aUser{
    string username=1;
    string password=2;
}

message isSuccess{
    bool success=1;
    string msg=2;
    string userID=3;
}

message session{
    bool success=1;
    string msg=2;
    string token=3;
    string timeLimit=4;
}